spec/data-types-and-values.html

```diff
@@ -7,7 +7,10 @@ <h1>ECMAScript Language Types</h1>
 
     <emu-clause id="sec-ecmascript-language-types-record-type">
       <h1>The Record Type</h1>
-      <p>TBD</p>
+      <p>The Record type is a coumpound primitive type. It is an ordered collection of primitive data properties.</p>
+      <p>A <dfn>primitive data property</dfn> associates a String key value with a primitive value or a coumpound primitive value called associated value.</p>

```

Comment:
I can see how you've derived this from the spec text for objects. I want to suggest that you take a more explicit approach, with Lists and Records to represent the contents, for a couple reasons:
- Juggling the notions of "the Record has a primitive data property" from "the Record wrapper has a data property" feels complicated. I'd prefer to think about just one property (the "virtual" one in the wrapper)
- This vague text raises more questions than it answers, since it it ambiguous about whether property attributes, ordering are supported.
- Newer constructs (like private fields and methods) end up having representations in terms of Records and Lists, and I'd like to continue this with newer things.

I would instead say, Record values have an associated List of Records of the form { [[Key]], [[Value]] }, where [[Key]] is a String and [[Value]] is any primitive (including compound primitives), which are sorted by [[Key]] in code unit order. Operations which create Record values are responsible for setting this up appropriately. The List of Records is never modified.

This PR does not contain any operation which creates new Record values. Maybe you could start with the definition of Record literals. Without an example like that, I can't really review how this definition works in practice.

Reply:
> This PR does not contain any operation which creates new Record values. Maybe you could start with the definition of Record literals. Without an example like that, I can't really review how this definition works in practice.

I will do that, in the meantime, in which section do you think this should appear?

Reply:
It's fine to put it here. We could consider moving it around when merging into the main spec.